Azure Private Link update (#2539)
astronomer/docs: astro/connect-azure.md

Astro clusters are pre-configured with the Azure blob private endpoint.

To request additional endpoints, or assistance connecting to other Azure services, you need to provide Astronomer support with the following information for the resource you want to connect using Private Link:

- Resource name
- Resource ID
- Private endpoint

As an example, let's assume you want to connect to Azure Container Registry, then follow these steps:

1. In your Azure portal, follow the [Azure documentation](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-get-started-portal?tabs=azure-cli) to create the [container registry](https://portal.azure.com/#create/Microsoft.ContainerRegistry). Copy the name of container registry.
2. Follow the [Azure documentation](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-private-link#create-a-private-endpoint---new-registry) to create a private endpoint for your container registry. Copy the name of the **Data endpoint** once private endpoint is created.

> 3. Then, from the left panel, go to **Overview** menu, and click on JSON view in **Essentials**, to copy the resource ID. You can also run Azure CLI command `az acr show -n myRegistry` to get the resource ID.

Reviewer: ```suggestion
3. Follow the [Azure documentation](https://learn.microsoft.com/en-us/azure-stack/user/container-registry-get-resource-id?view=azs-2301#get-the-resource-id-for-aks) to copy the resource ID for your registry. 
```

Author: @jwitz these are not the correct instructions for getting the JSON view for a container registry. The link assumes you have an Azure Stack Hub and directs you to go there, which is not part of the example and might confuse users.

> 4. Contact [Astronomer Support](https://support.astronomer.io/) and provide the resource name, private endpoint and resource ID in the ticket.

Reviewer: ```suggestion
4. Contact [Astronomer Support](https://cloud.astronomer.io/support) with your request to connect. Provide the resource name, data endpoint name, and resource ID.
```

> 5. When Astronomer support adds an Azure private endpoint, a corresponding private DNS zone and Canonical Name (CNAME) records are also created to allow you to address the service by its private link name. Astronomer support will send request for connection in Azure Portal's [Private Link Center](https://portal.azure.com/#view/Microsoft_Azure_Network/PrivateLinkCenterBlade/~/pendingconnections). 

Reviewer: ```suggestion
5. When Astronomer support adds an Azure private endpoint, corresponding private DNS zone and Canonical Name (CNAME) records are created to allow you to address the service by its private link name. Astronomer support will send the connection request in Azure Portal's [Private Link Center](https://portal.azure.com/#view/Microsoft_Azure_Network/PrivateLinkCenterBlade/~/pendingconnections). 
```

> 6. Once you approve the connection requests, acknowledge in the ticket. Astronomer support will then test whether the DNS resolves the endpoint correctly. They will confirm and update in the ticket. 

Reviewer: ```suggestion
6. Approve the connection requests from your Azure portal, then confirm that you've completed this in your support ticket. Astronomer support will then test whether the DNS resolves the endpoint correctly.
```

> 7. Now you can create Airflow connection and test your DAGs. In some circumstances, you might need to modify your DAGs to address the service by its private link name. For example, `StorageAccountA.privatelink.blob.core.windows.net` instead of `StorageAccountA.blob.core.windows.net`.

Reviewer: ```suggestion

After Astronomer configures the connection, you can create Airflow connections to your resource. In some circumstances, you might need to modify your DAGs to address the service by its private link name (For example, `StorageAccountA.privatelink.blob.core.windows.net` instead of `StorageAccountA.blob.core.windows.net`).
```

> Please note, you'll incur additional Azure infrastructure costs for every Azure private endpoint that you use. See [Azure Private Link pricing](https://azure.microsoft.com/en-us/pricing/details/private-link/).

Reviewer: ```suggestion
Note that you'll incur additional Azure infrastructure costs for every Azure private endpoint that you use. See [Azure Private Link pricing](https://azure.microsoft.com/en-us/pricing/details/private-link/).
```

Reviewer: This should go at the top
